plugin/ipam/cni.go

```diff
@@ -35,31 +36,76 @@ func (i *Ipam) Allocate(args *skel.CmdArgs) (types.Result, error) {
 	if containerID == "" {
 		return nil, fmt.Errorf("Weave CNI Allocate: blank container name")
 	}
-	var ipnet *net.IPNet
 
-	if conf.Subnet == "" {
-		ipnet, err = i.weave.AllocateIP(containerID, false)
+	var ipconfigs []*current.IPConfig
+
+	if len(conf.IPs) > 0 {
+		// configuration includes desired IPs
+
+		var ips []net.IP
+		for _, ip := range conf.IPs {
+			ipv4 := net.ParseIP(ip).To4()
+
+			if ipv4 == nil {
+				return nil, errors.New("allocation of ipv6 addresses is not implemented")
```

Comment:
can this error happen if the string is just invalid, like "banana" ?

Reply:
I added another error in case the provided value is not an IP.